### ISSUE TITLE
Error 504: Response Timeout - unable to download

### ISSUE THREAD

**Issue by jamesmachin** [trapias/TrelloExport#29]
Hello,

When using the TrelloExport to download data from one of our boards we get a red error box at the top right stating Error 504: Response Timeout

![1](https://user-images.githubusercontent.com/32671058/31382030-3ada68e2-adae-11e7-9493-df46006f18aa.png)

When we try open the downloaded file we get an error saying there was a problem with some content. If we click 'Yes' to recover it opens with just 3 columns of numbers.

![2](https://user-images.githubusercontent.com/32671058/31382115-83081416-adae-11e7-96d4-e537691aaf90.png)
![3](https://user-images.githubusercontent.com/32671058/31382121-87a08daa-adae-11e7-9f59-ef037c7ce449.png)

We think this may be due to our board having a lot of images uploaded. We are in the process of redownloading the images and making them smaller hoping it will work.

Any help or advice would be much appreciated :)

**Comment by trapias** [comment]
@jamesmachin do you have lots of cards too? I should improve loading actually.

**Comment by jamesmachin** [comment]
@trapias yes, the last time we downloaded the board we had around 550 cards currently open.
Is this something that could be sorted?

**Comment by trapias** [comment]
@jamesmachin yes I just cannot find the time to work on it 🙄 will update here ASAP!

**Comment by jamesmachin** [comment]
@trapias many thanks 👍

**Comment by jamesmachin** [comment]
@trapias is there a limit set to how many open cards the TrelloExtract can download? Or does it depend on the contents of each card?

**Comment by trapias** [comment]
@jamesmachin there’s a limit on the number of items you can get from Trello api, 1000 per call. I should modify the loading process, just cannot find the time 🙄

**Comment by jamesmachin** [comment]
@trapias Thanks for the info. No problem, let me know if this gets fixed so I know to download the upate

**Comment by Steph-CTS** [comment]
I believe that I am experiencing a related issue to this? I am unable to export anything though. 
![trelloexport error](https://user-images.githubusercontent.com/14115272/36752668-8c3c8828-1bb8-11e8-8f4f-c9615c10bff5.PNG)
I only get the red box that reads: "Board menu not open?"  We have very few cards on our board.

**Comment by trapias** [comment]
The "board menu not open" problem, as of issue #39 , is due to a bug - please use the Board side menu to open TrelloExport.

Solved with next release, coming soon.

**Comment by jamesmachin** [comment]
@trapias sorry I don't think this would be possible as we hold private information on our boards.
We haven't had this issue since the size of the board was reduced.

**Comment by craigbowman2** [comment]
Hi @trapias 

Have you received the other story that I raised earlier today?

**Comment by icarocastellani** [comment]
Hi @trapias,

First congratulations for your work, TrelloExport is the best app to extract data for Trello.

Unfortunately we are facing the issue - Error 504: Response Timeout after our Kanban reached more than 600 cards.

Do you have any predictions for a solution? I believe that this error is for a very small portion of users which does not make it such a priority.

**Comment by jiraghav** [comment]
HI @albertovelo, i have **move than 1k cards** in our card
I have also encountered the same problem but fixed by some changes

1. **To decrease the pageSize to 25** from 300 will fixed my problem.
2. On error this code will `readCards = -1` skip the cards and move on next list, i think you will need to **implement the retry logic in case on any api failure**.

Also one point to note: All board lists are not included, Fixed by **filter=all** in query parameters.

Thanks,
Hope you will do these changes

**Comment by trapias** [comment]
@jiraghav thanks for your help, will look into this within a few days.
Could you eventually share such a big Trello board with me, so that I can debug and make tests?
Thank you,
al.

**Comment by raghavji2050** [comment]
@trapias to provide access of board is not possible!, But my feedback really works.

**Comment by trapias** [comment]
@raghavji2050 I just committed version 1.9.7.3, could you make a test before I send it to the Chrome web store?

**Comment by trapias** [comment]
> @raghavji2050 I just committed version 1.9.7.3, could you make a test before I send it to the Chrome web store?

Didn't implement a retry logic yet

**Comment by jiraghav** [comment]
@trapias it's working fine. You can move forward.

**Comment by Schlesinger79** [comment]
Hi.  I 've been using the Trello Export for months with no issues until today.  Same issue with the response timeout and as a result it exports an incomplete file.  I'm using the latest version and still has the same issue.  I appreciate any help.

**Comment by trapias** [comment]
@Schlesinger79 sorry for this, I hardly can find the time to work on it - I should implement retry.
How many cards do you have on that dashboard?

**Comment by Schlesinger79** [comment]
Hi, thanks for your reply.  Actually its not so many cards and the have no attachments or any other complex item inside them.  I appreciate your help cause it's a brilliant extension that helped me and my company run a good dashboard on excel.

**Comment by Schlesinger79** [comment]
I forgot, its 540 cards on the board.

**Comment by trapias** [comment]
@Schlesinger79 thanks for your feedback, I’ll look into this as soon as possible - cannot tell when sorry 😞
